TEST_APPS/device/nfcapp/nfccommands.h

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) 2018 ARM Limited. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ * Licensed under the Apache License, Version 2.0 (the License); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef _NFCCOMMANDS_H_INCLUDED
+#define _NFCCOMMANDS_H_INCLUDED
+
+#if MBED_CONF_NFCEEPROM
+#include "nfcProcessEeprom.h"
+#else
+#include "nfcProcessCtrl.h"
+#endif
+
+extern events::EventQueue nfcQueue;
+
+// see https://support.microsoft.com/en-my/help/208427/maximum-url-length-is-2-083-characters-in-internet-explorer
+#define MAX_URL_LENGTH  2000
+
+/**
+ * HandleTestCommand turns all the typed-in/serial commands into function calls sent via a eventqueue to the driver
+ * shim/wrapper class. Most application level code that is agnostic of EEPROM/controller drivers lives in this class.
+ * Methods with cmd_ prefix map to the serial commands, and are building blocks for test cases and scenarios. The
+ * first function a test must call is typically the initnfc command. Commands report back a test verdict, and a
+ * NFC status code. The test verdict is always success unless the command is not allowed. Tests much check the
+ * NFC error code for NFC_OK or zero; this pattern allows us to write negative tests which expect a specific NFC error.
+ */
+class HandleTestCommand {
+public:
+    HandleTestCommand();
+    /** set corresponding mask bit on in referenced structure, return false if the supplied string cannot parse */
+    static bool set_protocol_target(mbed::nfc::nfc_rf_protocols_bitmask_t &bitmask, const char *protocolName);
+
+    /** return and clear the last result code. Type "help getlastnfcerror" for a list of error codes */
+    static int cmd_get_last_nfc_error(int argc, char *argv[])
```

Comment:
Can it be refactored so that:
- A pointer to `nfcQueue` is passed in the constructor instead of using `extern` above
- methods here are not static
- implementation of these methods live in the `.cpp` file

Reply:
- extern refactored
- 👎   statics are there because the test framework is 'C' at this time.
- implementations moved